src/driver/capacitor/CapacitorDriver.ts

```diff
@@ -74,22 +74,33 @@ export class CapacitorDriver extends AbstractSqliteDriver {
      * Creates connection with the database.
      */
     protected async createDatabaseConnection() {
+        const isDatabaseEncryted = this.options.encrypted || false;
+        const databaseMode = this.options.mode || "no-encryption";
+        const databaseVersion =
+            typeof this.options.version === "undefined"
+                ? 1
+                : this.options.version;
         const connection = await this.sqlite.createConnection(
             this.options.database,
-            false,
-            "no-encryption",
-            1
+            isDatabaseEncryted,
+            databaseMode,
+            databaseVersion
         );
         await connection.open();
 
         // we need to enable foreign keys in sqlite to make sure all foreign key related features
         // working properly. this also makes onDelete to work with sqlite.
-        await connection.query(`PRAGMA foreign_keys = ON`, []);
-
-        if (this.options.journalMode) {
-            await connection.query(`PRAGMA journal_mode = ?`, [
-                this.options.journalMode,
-            ]);
+        await connection.query(`PRAGMA foreign_keys = ON`);
+
+        if (
+            this.options.journalMode &&
+            ["DELETE", "TRUNCATE", "PERSIST", "MEMORY", "WAL", "OFF"].indexOf(
+                this.options.journalMode
+            ) !== -1
+        ) {
+            await connection.query(
+                `PRAGMA journal_mode = ${this.options.journalMode}`
```

Comment:
Why aren't we using parameters here?

Reply:
sqlite unfortunately does not allow parameters in PRAGMA queries/requests (that's what i tried before)

Reply:
Got it - I see that in the sqlite stuff as well.

Can you move this change to a separate PR so we can have a better changelog & commit history?

Reply:
done: https://github.com/typeorm/typeorm/pull/7873